lib/spack/spack/cmd/checksum.py

```diff
@@ -142,7 +142,7 @@ def checksum(parser, args):
                 *v_match.group(1, 2, 3, 4),
                 ",\n",
                 re.sub(r".", " ", v_match.group(1)),
-                'url="{0}"'.format(vstring_url_dict[url]),
+                'url="{0}"'.format(vstring_url_dict[matched_version]),
```

Comment:
I think this might be better as
```suggestion
                'url="{0}"'.format(url),
```
which I think was my/the original intent.